### ISSUE TITLE
cache - forks

### ISSUE THREAD

**Issue by kumavis** [MetaMask/provider-engine#24]
from the chats:

 `eth_getTransactionByHash` should be fork type not perma, now that i think of it
bc if the block the tx was included in is "forked out", the tx will settle in another block
but have the same tx hash

its quite an edge case but `eth_getCode` result could also change if the contract constructor included something like a blockNumber or coinbase opcode and there was a fork

**Comment by kumavis** [comment]
blocked by MetaMask/eth-block-tracker#30

**Comment by kumavis** [comment]
duplicate of #29
